### PR TITLE
Add Litl keyboard

### PR DESCRIPTION
**Pull request by mohoyt**
## Description

Litl is a 40% easy to build through hole keyboard that I have designed and built. This PR adds the keyboard and default keymap to QMK. 

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/sthlmkb/litl/info.json [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    "diode_direction": "ROW2COL",
+    "features": {
+         "bootmagic": false,
+         "command": false,
+         "console": false,
+         "extrakey": true,
+         "debug": false,
+         "mousekey": false,
+         "encoder": true,
+         "oled_driver": true,
```
Comment (zvecr) [inline keyboards/sthlmkb/litl/info.json:16]:
```suggestion
```

File: keyboards/sthlmkb/litl/config.h [hunk 2]
```diff
@@ -0,0 +1,40 @@
+/*
```
Comment (zvecr) [inline keyboards/sthlmkb/litl/config.h:1]: This file should be removed

File: keyboards/sthlmkb/litl/rules.mk [hunk 3]
```diff
@@ -0,0 +1 @@
+
```
Comment (zvecr) [inline keyboards/sthlmkb/litl/rules.mk:1]:
```suggestion
# This file intentionally left blank
```

**Review by zvecr — CHANGES_REQUESTED** [review]

File: keyboards/sthlmkb/litl/config.h (hunk 2 above)
Comment (mohoyt) [reply keyboards/sthlmkb/litl/config.h]: Done!

File: keyboards/sthlmkb/litl/info.json [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+    },
+    "matrix_pins": {
+         "cols": ["F6", "F7", "B1", "B3", "B2", "B6"],
+         "rows": ["B4", "B5", "D4", "C6", "D7", "E6", "F5", "F4"]
+     },
+    "processor": "atmega32u4",
+    "usb": {
+         "device_version": "1.0.0",
+         "pid": "0x5353",
+         "vid": "0x1729"
```
Comment (drashna) [inline keyboards/sthlmkb/litl/info.json:26]: This vendor ID is in use.  Please change.  Anything about 0x4000 is generally okay, but the higher, the better. 

https://www.the-sz.com/products/usbid/index.php?v=0x1729&p=&n=

File: keyboards/sthlmkb/litl/keymaps/default/keymap.c [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+    if (clockwise) {
+      tap_code_delay(KC_VOLU, 10);
+    } else {
+      tap_code_delay(KC_VOLD, 10);
+    }
+  /* You can copy the code and change the index for every encoder you have. Most
+     keyboards will only have two, so this piece of code will suffice. */
+  }
+  return false;
+}
```
Comment (drashna) [inline keyboards/sthlmkb/litl/keymaps/default/keymap.c:60]: if anything, you may want to use encoder map here.
```suggestion
```

File: keyboards/sthlmkb/litl/litl.c [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#include "litl.h"
```
Comment (drashna) [inline keyboards/sthlmkb/litl/litl.c:16]: This allows for a default behavior to be defined, without the need for the user to specify the behavior.  This is very useful for the configurator which doesn't have a way to configure this behavior, currently.

```suggestion
#include "litl.h"

#ifdef ENCODER_ENABLE
bool encoder_update_kb(uint8_t index, bool clockwise) {
    if (!encoder_update_user(index, clockwise)) { return false; }
    if (index == 0) {
        if (clockwise) {
            tap_code_delay(KC_VOLU, 10);
        } else {
            tap_code_delay(KC_VOLD, 10);
        }
    }
    return true;
}
#endif
```

File: keyboards/sthlmkb/litl/info.json [hunk 7]
```diff
@@ -0,0 +1,50 @@
+{
+    "manufacturer": "STHLM kb",
+    "keyboard_name": "litl",
+    "url": "https://github.com/mohoyt/litl",
+    "maintainer": "mohoyt",
+    "bootloader": "caterina",
```
[the remark this answers is not in the document]
Comment (mohoyt) [reply keyboards/sthlmkb/litl/info.json]: Fixed

File: keyboards/sthlmkb/litl/info.json (hunk 4 above)
Comment (mohoyt) [reply keyboards/sthlmkb/litl/info.json]: Updated to a unique vendor ID. Thanks!

**Review by drashna — APPROVED** [review]

**Review by fauxpark — APPROVED** [review]

**Pull request merged**